rfcs/xxxx-authorization/text.md

```diff
@@ -0,0 +1,196 @@
+The time should be chosen so that the check only needs to be done occasionally.  
+Probably anywhere from 30 seconds to 30 minutes is fine. 
+
+
+#### Handling of obfuscated datasets
+
+This discussion assumes that the obfuscated path has this form:  `/e/<obfuscated_uuid>/<datset_name>
+.cxg`, although other options are possible.
+
+In this way, it is easy to separate out the two components from the path.
+
+For obfuscated datasets, the authorization portion will go through an API to check the
+obfuscated_uuid and dataset.  It will ensure that the obfuscated_uuid belongs to a collection
+that contains the dataset, and it will ensure that the obfuscated_uuid has not expired.
+  
+`GET /v1/dataset/access_obfuscated`
+
+**Description:** determine if the obfuscated_uuid provides access to the dataset
+
+**Parameters:**
+
+1. obfuscated_uuid [REQUIRED]:  the collection's obfuscated_uuid
+2. dataset_name [REQUIRED]: the name of the dataset (e.g. "pbmc3k.cxg")
+
+**Response:**
+
+1. 200 OK, if the obfuscated_uuid is valid (and not expired) obfuscated_uuid for the
+ collection containing the dataset
+2. 401 Unauthorized
+3. 404 dataset not found or obfuscated_uuid not found.
+
+#### Community releaseas vs hosted cellxgene.
+
+There should be a configuration option to set the authorization type, such as how we do with
+authentication.  The authorization type we use for CZI hosted cellxgene possibly may be better
+implemented as a plugin, using our plugin architecture, so as not to make public the
+details of the implementation.
+
+In the scenario described in this document, the authorization component will need to answer the
+question: does the current user have access to the dataset at this path?  It will also need to
+return the dataset name from the path, since in the case of the obfuscated_uuid feature, the name 
+is embedded in the path, and needs to be extracted.  
+ 
+#### Caching issues
+
+Checking authorization requires access to the cellxgene explorer backend, which accesses the
+cellxgene portal backend.
+ 
+I need to do some research to determine how to force the authorization check on each API request
+to the explorer, but also caches copies of the data on cloudfront.
+
+### cellxgene Portal
+
+The `/v1/dataset/access/` and `v1/dataset/access_obfuscated` APIs will be implemented in the portal
+ code base, since they will need access to the portal database.
+
+To implement this efficiently, it may be useful to create two database views: One that maps
+`dataset.name` to `collection.owner` and `collection.visibility`.  And one that maps
+ `obfuscated_uuid` and `dataset.name` to `collection.obfuscated_uuid_expiration`.
+ 
+Note, `obfuscated_uuid_expiration` is not currently part of the database.
+
+
+### UX issues
+
+#### portal
+
+1. Can the owner create new obfuscated_uuid's, and set an expiration?
+1. The owner will need to retrieve the obfuscated URL of the collection, so it can be shared.
+1. Can there be more than one obfuscated_uuid?
```

Comment:
Not at this time. There is one bearer token shared with any reviewer or collaborator. (In contrast to a more finer grained complex [approach](https://github.com/broadinstitute/single_cell_portal/wiki/Sharing-Private-Studies).